### PR TITLE
Add health check endpoint

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Example response:
```
{
  "status": "ERROR",
  "checks": [
    {
      "name": "registration",
      "status": "OK"
    },
    {
      "name": "commandConnection",
      "status": "OK"
    },
    {
      "name": "metricsConnection",
      "status": "ERROR"
    }
  ]
}
```

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: docs/swagger.json [hunk 1]
```diff
@@ -195,6 +218,44 @@
       },
       "x-go-package": "github.com/nginx/agent/v2/src/plugins"
     },
+    "HealthResponse": {
+      "type": "object",
+      "properties": {
+        "checks": {
+          "description": "Array of health checks",
+          "type": "array",
```
[the remark this answers is not in the document]
Comment (olli-holmala) [reply docs/swagger.json:226]: Is it just meant for sending out more granular info on what the status is of different services/plugins in the Agent?

File: docs/swagger.json [hunk 2]
```diff
@@ -195,6 +218,44 @@
...
+      "x-go-package": "github.com/nginx/agent/v2/src/plugins"
+    },
+    "HealthStatusCheck": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Health check name",
+          "type": "string",
+          "x-go-name": "Name",
+          "example": "commandServiceConnection"
```
[the remark this answers is not in the document]
Comment (dhurley) [reply docs/swagger.json:248]: wasn't sure what name to give for us to say that we are able to still send and receive commands over the gRPC stream. Would `commandServiceConnectivity` be better?

File: src/plugins/agent_api.go [hunk 3]
```diff
@@ -605,6 +653,94 @@ func (h *NginxHandler) syncProcessInfo(processInfo []*core.Process) {
...
+	checks := []HealthStatusCheck{}
+
+	registrationStatus := okStatus
+	commandServiceStatus := okStatus
+	metricsServiceStatus := okStatus
+
+	if rh.config.IsGrpcServerConfigured() {
+		if !rh.isGrpcRegistered {
+			registrationStatus = errorStatus
+			overallStatus = errorStatus
```
Comment (olli-holmala) [inline src/plugins/agent_api.go:699]: Some of these could potentially be simplified to one line.
```suggestion
			registrationStatus, overallStatus = errorStatus, errorStatus
```

**Review by Dean-Coakley — APPROVED** [review]

File: docs/swagger.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliveromahony) [reply docs/swagger.json:248]: `command` or `commandConnection` would I think?

[the remark this answers is not in the document]
Comment (oliveromahony) [reply docs/swagger.json:248]: same change should be applied to metrics

[the remark this answers is not in the document]
Comment (dhurley) [reply docs/swagger.json]: updated

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**